src/test/scala/io/iohk/ethereum/ledger/BlockExecutionSpec.scala

```diff
@@ -89,6 +90,23 @@ class BlockExecutionSpec extends AnyWordSpec with Matchers with ScalaCheckProper
         blocks.head.block shouldBe block1
         error.isDefined shouldBe true
       }
+
+      "block with checkpoint and without txs" in new BlockchainSetup {
+        val checkpoint = ObjectGenerators.fakeCheckpointGen(2, 5).sample.get
+        val blockWithCheckpoint = CheckpointingTestHelpers.createBlockWithCheckpoint(validBlockParentHeader, checkpoint)
+
+        val mockValidators = MockValidatorsAlwaysSucceed
+        val newConsensus: TestConsensus = consensus.withVM(vm).withValidators(mockValidators)
+        val blockValidation = new BlockValidation(newConsensus, blockchain, BlockQueue(blockchain, syncConfig))
+        val blockExecution =
+          new BlockExecution(blockchain, blockchainConfig, newConsensus.blockPreparator, blockValidation)
+
+        val (blocks, error) = blockExecution.executeBlocks(List(blockWithCheckpoint), defaultBlockHeader.difficulty)
+
+        blocks.size shouldBe 1
+        blocks.head.block shouldBe blockWithCheckpoint
```

Comment:
Should we check that world state is unchanged? Can we do that here?

Reply:
we can't as all intermediate world states are created and hidden inside components.
But we know that state is unchanged because the previous failing state root validation is running and passing normally (it is not mocked, `MockValidatorsAlwaysSucceed` is the validator of header and body)

Reply:
> we can't as all intermediate world states are created and hidden inside components.

But blockchain is not mocked and accessible here. Could we use that?

> But we know that state is unchanged because the previous failing state root validation is running and passing normally (it is not mocked, `MockValidatorsAlwaysSucceed` is the validator of header and body)

I'm not quite sure which part of the code you are referring to